core/types/transaction_marshalling.go

```diff
@@ -470,6 +485,23 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 			return errors.New("missing required field 'sourceHash' in transaction")
 		}
 		itx.SourceHash = *dec.SourceHash
+		// IsSystemTx may be omitted. Defaults to false.
+		if dec.IsSystemTx != nil {
+			itx.IsSystemTransaction = *dec.IsSystemTx
+		} else {
+			// [Kroma: START]
+			inner = &KromaDepositTx{
+				SourceHash: itx.SourceHash,
+				From:       itx.From,
+				To:         itx.To,
+				Mint:       itx.Mint,
+				Value:      itx.Value,
+				Gas:        itx.Gas,
+				Data:       itx.Data,
+			}
+			// [Kroma: END]
+		}
+
 		if dec.Nonce != nil {
 			inner = &depositTxWithNonce{DepositTx: itx, EffectiveNonce: uint64(*dec.Nonce)}
 		}
```

Comment:
I think we need kromaDepositTxWithNonce also. If user deposit tx comes in, dec.Nonce will not be nil and inner will be set not using KromaDepositTx, right?